### PR TITLE
E2E tests: test ingress for externalip services

### PR DESCRIPTION
**Pull request by fedepaol**
**- What this PR does and why is it needed**

 We add a new test that creates an externalip service and tries to reach all the endpoints (one for each node) from an external container. The external container goes over all the ips assigned to the node and tries the connectivity through each one of them.

**- Special notes for reviewers**

This depends on https://github.com/ovn-org/ovn-kubernetes/pull/1985 and includes its commits.
Only the last commit is related to this change

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -30,6 +34,10 @@ const (
 	ciNetworkName        = "kind"
 )
 
+var (
+	agnhostImage = imageutils.GetE2EImage(imageutils.Agnhost)
```
Comment (aojea) [inline test/e2e/e2e_test.go:38]: I start to think that we should hardcode and not depend on the framework, what do you think?
```
k8s.gcr.io/e2e-test-images/agnhost:2.26
```

you already felt the pain of dealing with these images upstream

File: test/e2e/e2e_suite_test.go [hunk 2]
```diff
@@ -40,7 +39,7 @@ func TestMain(m *testing.M) {
 	// Now that we know which Viper config (if any) was chosen,
 	// parse it and update those options which weren't already set via command line flags
 	// (which have higher priority).
-	if err := viperconfig.ViperizeFlags(*viperConfig, "e2e", flag.CommandLine); err != nil {
+	if err := viperizeFlags(*viperConfig, "e2e", flag.CommandLine); err != nil {
```
Comment (aojea) [inline test/e2e/e2e_suite_test.go:42]: I think that we can just drop it, we don´t use it for these tests

Comment (fedepaol) [reply test/e2e/e2e_suite_test.go:42]: Ah sweet, I was not sure about this.

**Comment by fedepaol** [comment]
@aojea will address the comments in https://github.com/ovn-org/ovn-kubernetes/pull/1990 which is the base for this pr

**Review by trozet — APPROVED** [review]
thanks @fedepaol

**Pull request merged**